File: intelligencecity_manager_interface/src/main/java/com/shanghaiuniversity/manager/service/RubbishpicService.java
package com.shanghaiuniversity.manager.service;

import com.ShanghaiUniversity.entity.PageResult;
import com.ShanghaiUniversity.pojo.Rubbishpic;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface RubbishpicService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<Rubbishpic> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加
     */
    public void add(Rubbishpic rubbishpic);

    /**
     * 修改
     */
    public void update(Rubbishpic rubbishpic);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public Rubbishpic findOne(Long id);

    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Rubbishpic rubbishpic, int pageNum, int pageSize);

}

File: intelligencecity_manager_web/src/main/webapp/assets/js/service/rubbishpicService.js
//服务层
app.service('rubbishpicService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../rubbishpic/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../rubbishpic/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../rubbishpic/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../rubbishpic/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../rubbishpic/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../rubbishpic/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../rubbishpic/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
